src/map/script.cpp

```diff
@@ -11337,18 +11337,28 @@ BUILDIN_FUNC(sc_end)
 
 /**
  * Ends all status effects from any learned skill on the attached player.
- * sc_end_class {<char_id>};
+ * if <job_id> was given it will end all the effect for that job
+ * if <job_id> is 0 it will return the job of the invoking character
+ * sc_end_class {<job_id>{,<char_id>}};
  */
 BUILDIN_FUNC(sc_end_class)
 {
 	struct map_session_data *sd;
 	uint16 skill_id;
-	int i;
+	int class_ = 0;
```

Comment:
Remove init here

Reply:
here i didn't understand what to do

Reply:
He means remove the `= 0` part. Just define the variable: `int class_;`